src/iexamine.cpp

```diff
@@ -4511,8 +4515,12 @@ void iexamine::ledge( player &p, const tripoint &examp )
                     } else {
                         query = _( "You may have problems climbing back up.  Climb down?" );
                     }
-                } else {
+                } else if( has_grapnel ) {
                     query = _( "Use your grappling hook to climb down?" );
+                } else if ( p.has_trait( trait_WEB_RAPPEL ) ) {
+                    query = _( "Spin a dragline to climb down?" );
+                } else if ( p.has_trait( trait_VINES2 ) || p.has_trait( trait_VINES3 ) ) {
+                    query = _( "Climb down with one of your vines?" );
                 }
 
                 if( !query_yn( query.c_str() ) ) {
```

Comment:
You're triggering a warning here because query is no longer guaranteed to be initialized.

Reply:
```
                } else if ( p.has_trait( trait_WEB_RAPPEL ) ) {
                    query = _( "Spin a dragline to climb down?" );
                } else if ( p.has_trait( trait_VINES2 ) || p.has_trait( trait_VINES3 ) ) {
                    query = _( "Climb down with one of your vines?" );
                } else {
                    query = _( "Use your grappling hook to climb down?" );
                }
```

Would something like that work, or?

Reply:
```
                } else if ( p.has_trait( trait_WEB_RAPPEL ) ) {
                    query = _( "Spin a dragline to climb down?" );
                } else if ( p.has_trait( trait_VINES2 ) || p.has_trait( trait_VINES3 ) ) {
                    query = _( "Climb down with one of your vines?" );
                } else {
                    query = _( "Use your grappling hook to climb down?" );
                }
```

@kevingranade Would something like that work, or...?